app/app.test.ts

```diff
@@ -1,15 +1,54 @@
 import app from './app';
 import supertest from 'supertest';
 
-const request = supertest(app);
-
 describe('App Integration', () => {
   describe('GET /listener', () => {
     it('returns OK', async () => {
+      const request = supertest(app);
       const response = await request.get('/listener');
 
       expect(response.status).toBe(200);
-      expect(response.text).toBe('OK');
+      expect(response.body).toEqual({ message: 'OK' });
+    });
+  });
+
+  describe('Handlers', () => {
+    describe('Error', () => {
+      describe('bodyParserHandler', () => {
+        it('returns error when JSON payload is malformed', async () => {
+          const request = supertest(app);
```

Comment:
Move `supertest` init so that all tests use the same app; Check weather it doesn't interfere with parallel test runs